cilium-cli/connectivity/builder/egress_gateway_with_l7_policy.go

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package builder
+
+import (
+	_ "embed"
+
+	"github.com/cilium/cilium/cilium-cli/connectivity/check"
+	"github.com/cilium/cilium/cilium-cli/connectivity/tests"
+	"github.com/cilium/cilium/cilium-cli/utils/features"
+	"github.com/cilium/cilium/pkg/versioncheck"
+)
+
+//go:embed manifests/client-egress-icmp.yaml
+var clientEgressICMPYAML string
+
+//go:embed manifests/client-egress-l7-http-external-node.yaml
+var clientEgressL7HTTPExternalYAML string
+
+type egressGatewayWithL7Policy struct{}
+
+func (t egressGatewayWithL7Policy) build(ct *check.ConnectivityTest, templates map[string]string) {
+	newTest("egress-gateway-with-l7-policy", ct).
+		WithCondition(func() bool {
+			return versioncheck.MustCompile(">=1.16.0")(ct.CiliumVersion) && ct.Params().IncludeUnsafeTests
+		}).
+		WithCiliumPolicy(clientEgressICMPYAML).
+		WithCiliumPolicy(clientEgressOnlyDNSPolicyYAML).               // DNS resolution only
```

Comment:
nit: This change in the go.mod gets reverted immediately after in the next commit, could fold this into the original commit:

```suggestion
		WithCiliumPolicy(clientEgressOnlyDNSPolicyYAML).               // DNS resolution only
```

Reply:
which go.mod change is this? looks like this is pointing a different file 👀

Reply:
This line gets changed in the first go.mod commit, then changed back because it broke linter in the next commit. No need to follow up on this.